tests/neg/higher-order-quoted-expressions.scala

```diff
@@ -0,0 +1,44 @@
+// Draft of higher order expressions
+// ---------------------------------
+// A higher order expression is a an expression that may contain unbound variables.
+// Unbound variables are listed as arguments of the type lambda of the expression.
+// Splicing can only be done on Expr[_ <: Any].
+// `apply methods on higher order expression provide a way to replace the unbound variables.
+
+// This can be used for HOAS quoted patterns to not return a lambda.
+// The use would be the same as each higher order expression would have an apply method.
+// But as it would be an expression, the expression would showable.
+// The expression could also be directly transformed into a Term of the reflection API.
```

Comment:
You mean, we could have a Term with holes? How would that work?

Reply:
Not no holes in the sense of the `Hole` AST. It would contain references to unbound variables in the current scope. We should probably also have a way to list the symbols of those variables to be able to identify them when reflecting on the AST. I still need to figure out what is the best way to encode this internally.

Reply:
>  It would contain references to unbound variables in the current scope. 

So what can we do with such a term? I thought translating from an Expr to a Term should always produce a valid hygiienic tree.